sympy/solvers/solveset.py

```diff
@@ -2139,8 +2361,26 @@ def solveset(f, symbol=None, domain=S.Complexes):
         f = f.xreplace({d: e})
     f = piecewise_fold(f)
 
-    return _solveset(f, symbol, domain, _check=True)
+    result = _solveset(f, symbol, domain, _check=True)
+    if _is_lambert(f, symbol) and isinstance(result,FiniteSet):
+        # it could be extended to other than lambert
```

Comment:
It shouldn't be necessary to make a special case for the lambert solver here in the main solveset function